src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -439,7 +452,16 @@ public Mutex GetArchiveMutex(string fileName)
                 var fileInfo = new FileInfo(filePath);
                 if (fileInfo.Exists)
                 {
-                    return Time.TimeSource.Current.FromSystemTime(fileInfo.GetCreationTimeUtc());
+                    result = fileInfo.GetCreationTimeUtc();
+                    if (result.Value.Year < 1980)
```

Comment:
could we get rid of the code duplication?

Maybe something like `useFallBack(dateTime r) => r.Year < 1980 && !PlatformDetector.IsDesktopWin32` ?